int32 time_to_ckpt
